core/tee/tee_obj.c

```diff
@@ -47,6 +47,8 @@ TEE_Result tee_obj_get(struct tee_ta_ctx *ctx, uint32_t obj_id,
 	TAILQ_FOREACH(o, &ctx->objects, link) {
 		if (obj_id == (uint32_t) o) {
 			*obj = o;
+			if (o->info.objectType == TEE_TYPE_CORRUPTED_OBJECT)
```

Comment:
This function is supposed to only translate a object handle into a pointer to a 'struct tee_obj'. This extra "feature" will break at least `tee_svc_cryp_obj_close()`